src/nvc.rs

```diff
@@ -2,6 +2,100 @@ use instruction::*;
 use interconnect::*;
 
 use std::collections::HashSet;
+use std::fmt;
+
+#[derive(Copy, Clone)]
+pub struct NvcCacheEntry {
+    pub tag: u32,
+    pub base_addr: u32,
+    pub subblock_valid: [bool; 2],
+}
+
+impl NvcCacheEntry {
+    fn new() -> NvcCacheEntry {
+        NvcCacheEntry {
+          tag: 0,
+          base_addr: 0,
+          subblock_valid: [false; 2],
+        }
+    }
+}
+
+impl fmt::Display for NvcCacheEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "base_addr: 0x{:08x}, tag: 0x{:08x}, sb1 valid: {:5}, sb2 valid: {:5}", self.base_addr, self.tag, self.subblock_valid[0], self.subblock_valid[1])
+    }
+}
+
+pub struct NvcCache {
+    hits: usize,
+    misses: usize,
+    cache_entries: [NvcCacheEntry; 128],
+}
+
+impl NvcCache {
```

Comment:
nit: I think this should be called `Cache` without the `Nvc` prefix. This would make the fully qualified name `nvc::Cache` rather than `nvc::NvcCache`.